web/modules/custom/nys_bills/src/Plugin/Block/AmendmentsBlock.php

```diff
@@ -175,14 +176,29 @@ public function build() {
         $prev_vers_pre = $this->t('Versions Introduced in @current Legislative Session:', ['@current' => $current]);
       }
 
+      // Create $amended_versions_tab_data array.
+      $amended_versions_tab_data = [];
+      foreach ($amendments as $amendment_title => $amendment_array) {
+        $amendment_node = $amendment_array['node'];
+        if ($amendment_node instanceof Node) {
+          try {
+            $amendment_node_url = $amendment_node->toUrl()->toString();
+          }
+          catch (\Exception $e) {
+            $amendment_node_url = '/node/' . $amendment_node->id();
```

Comment:
Having the fallback is a good idea, but we still want to avoid the `/node/xxx` style of URL.  What other ideas do we have here?

Reply:
@routinet, `$amendment_node->toUrl()->toString()` will always return the aliased path, as long as it exists (which is the responsibility of the `validateAliases()` method we have been separately inspecting). Note that `toString()` will actually return the `/node/xxx` path if an alias is not found, so let me know if that internal path should be avoided in all cases for some reason (however, I can't think why, considering the node ID is already all over the generated HTML). I'm not aware of anything else we can do to avoid the `/node/xxx` path being used, if an alias is missing.

Note: after realizing `toString()` will already fall back on the `/node/xxx` path, I've simplified the exception handling to simply return a empty URL (in what is really an exceptional case--a node without a NID).